### PR TITLE
adding the web app foundation code to build on

### PR DESCRIPTION
**Pull request by wsedor**
# Description

- Adds docker container structure for the web app using https://github.com/docker/awesome-compose/tree/master/react-express-mongodb as the sample code. This is inline with the webapp tech stack document https://docs.google.com/document/d/1L4WijvqfePeJhPPAo68qD6ndY1rVgwe5WLyGyJVpvDE/edit?usp=sharing
- Code spins up 3 docker containers: frontend (React), backend (NodeJS) and database (mongodb)
- frontend is served at `localhost:3000` and displays basic User Interface

Note: The sample code utilizes the React ToDo example so some of these are within this PR. As the webapp works moves along these will be removed accordingly.

# Testing

Tested basic functionality. From the web-app folder, run `docker compose up -d`.

See screenshots of output below

#### List of running docker containers
image
image

#### Frontend view served by frontend container
image

# Reviewers
@KutuDev
@jojoboateng

### CODE REVIEW

File: webapp/.docker/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+services:
```
Comment (wsedor) [inline webapp/.docker/docker-compose.yaml:1]: this file contains the directives that spin up the docker containers. See the README file for detailed explanation 
https://github.com/nsmq-ai/nsmqai/pull/7/files#diff-91584b86758870da3cf7b93afcb7ad605946bafc5f1cb0de288d16a9f55f8552R1

File: webapp/backend/models/todos/todo.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// TODO: Convert code in this file over to NSMQ AI database models
```
Comment (wsedor) [inline webapp/backend/models/todos/todo.js:1]: the sample code uses the React ToDo example. As the webapp work is being done these areas will be removed acordingly

File: webapp/backend/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// TODO: Convert code in this file over to NSMQ AI backend routes
```
Comment (wsedor) [inline webapp/backend/routes/index.js:1]: the sample code uses the React ToDo example. As the webapp work is being done these areas will be removed accordingly

File: webapp/frontend/src/App.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  componentDidMount() {}
+
+  render() {
+    return (
+      <div className="App container">
+        <div className="container-fluid">
+          <div className="row">
+            <div className="col-xs-12 col-sm-8 col-md-8 offset-md-2">
+              <h1>NSMQ AI</h1>
```
Comment (wsedor) [inline webapp/frontend/src/App.js:17]: using basic strings here. once frontend work is underway these will be moved to a properties file

File: webapp/frontend/src/components/AddTodo.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+// TODO: Replace code in this file over to NSMQ AI frontend components
```
Comment (wsedor) [inline webapp/frontend/src/components/AddTodo.js:1]: the sample code uses the React ToDo example. As the webapp work is being done these areas will be removed accordingly.

File: webapp/frontend/src/components/TodoList.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+// TODO: Replace code in this file over to NSMQ AI frontend components
```
Comment (wsedor) [inline webapp/frontend/src/components/TodoList.js:1]: the sample code uses the React ToDo example. As the webapp work is being done these areas will be removed accordingly

**Comment by KutuDev** [comment]
lgtm

File: webapp/README.md [hunk 7]
```diff
@@ -1,3 +1,133 @@
-# webapp
-webapp team works on:
-- Web app that integrates different ML systems, streams contest video from YouTube / Facebook or live speech input, and output answers via text and speech in < 1 sec
+## NSMQ AI Web application
```
[the remark this answers is not in the document]
Comment (wsedor) [reply webapp/README.md:1]: resolving comment as changes have been applied

**Pull request merged**